src/react/useLock.ts

```diff
@@ -0,0 +1,55 @@
+import { useContext, useEffect, useState } from 'react';
+import { SpaceContext } from './contexts/SpaceContext.js';
+
+import type { LockStatus, SpaceMember, Lock } from '../types.js';
+
+interface UseLockResult {
+  status: LockStatus | null;
+  member: SpaceMember | null;
+}
+
+/*
+ * Returns the status of a lock and, if it has been acquired, the member holding the lock
+ */
+export function useLock(lockId: string): UseLockResult {
+  const space = useContext(SpaceContext);
```

Comment:
I thought we're going to add a `ref` as an argument so that we knew what component to lock...?